.github/workflows/run-server-tests.yml

```diff
@@ -5,8 +5,10 @@ on:
   pull_request:
     branches:
       - main
+    paths:
+      - server/**
```

Comment:
Now the server tests only run when something inside the directory `server` changes. Otherwise this check will be skipped.